programs/compressed-token/src/burn.rs

```diff
@@ -0,0 +1,318 @@
+            if let Some(delegated_transfer) = inputs.delegated_transfer.as_ref() {
+                let mut vec = vec![false; 1];
+                vec[delegated_transfer.delegate_change_account_index as usize] = true;
+                (Some(vec), delegated_transfer.owner, Some(*authority))
+            } else {
+                (None, *authority, None)
+            };
+        let mut output_compressed_accounts =
+            vec![OutputCompressedAccountWithPackedContext::default(); 1];
+        create_output_compressed_accounts::<true, false>(
+            &mut output_compressed_accounts,
+            inputs.mint,
+            &[authority; 1],
+            delegate,
+            is_delegate,
+            &[change_amount],
+            None, // TODO: add wrapped sol support
+            &hashed_mint,
+            &[inputs.change_account_merkle_tree_index],
+        )?;
+        output_compressed_accounts
+    } else {
+        Vec::new()
+    };
+    add_token_data_to_input_compressed_accounts::<false>(
+        &mut compressed_input_accounts,
+        input_token_data.as_slice(),
+        &hashed_mint,
+    )?;
+    Ok((compressed_input_accounts, output_compressed_accounts))
+}
+
+#[cfg(not(target_os = "solana"))]
+pub mod sdk {
+
+    use anchor_lang::{AnchorSerialize, InstructionData, ToAccountMetas};
+    use light_system_program::{
+        invoke::processor::CompressedProof, sdk::compressed_account::MerkleContext,
+    };
+    use solana_sdk::{instruction::Instruction, pubkey::Pubkey};
+
+    use crate::{
+        process_transfer::{
+            get_cpi_authority_pda,
+            transfer_sdk::{
+                create_input_output_and_remaining_accounts, to_account_metas, TransferSdkError,
+            },
+            DelegatedTransfer,
+        },
+        token_data::TokenData,
+    };
+
+    use super::CompressedTokenInstructionDataBurn;
+
+    pub struct CreateBurnInstructionInputs {
+        pub fee_payer: Pubkey,
+        pub authority: Pubkey,
+        pub root_indices: Vec<u16>,
+        pub proof: CompressedProof,
+        pub input_token_data: Vec<TokenData>,
+        pub input_merkle_contexts: Vec<MerkleContext>,
+        pub change_account_merkle_tree: Pubkey,
+        pub mint: Pubkey,
+        pub burn_amount: u64,
+        pub signer_is_delegate: bool,
+    }
+
+    pub fn create_burn_instruction(
+        inputs: CreateBurnInstructionInputs,
+    ) -> Result<Instruction, TransferSdkError> {
+        let (remaining_accounts, input_token_data_with_context, _) =
+            create_input_output_and_remaining_accounts(
+                &[inputs.change_account_merkle_tree],
+                &inputs.input_token_data,
+                &inputs.input_merkle_contexts,
+                &inputs.root_indices,
+                &Vec::new(),
+            );
+        let outputs_merkle_tree_index = remaining_accounts
+            .get(&inputs.change_account_merkle_tree)
+            .unwrap();
```

Comment:
`?`

Especially here, since it's an SDK code, so we are going to treat it as a library.